index.rst

```diff
@@ -10,16 +10,200 @@
+
+As of May 2023, the Prompt Processing system starts processing when it receives a ``next_visit`` event from the summit before observations are taken.
+The event is received by a fan-out service, which forwards a derived event for each detector to a pool of workers.
+Each fanned-out event represents the exposure(s) for a given visit with a given detector.
+Because the workers handle their assigned exposures independently and in parallel, this framework is efficient and flexible with respect to unexpected computational demands.
+However, it's impossible to make assumptions about the order of events between workers, and any given worker may be created or destroyed at any time.
+
+Each of the several hundred workers maintains an internal ("local") Butler repository for its processing; this keeps the central repository from becoming a performance bottleneck for pipeline execution.
+On receiving its visit-detector assignment, each worker downloads any calibrations, templates, etc. it will need but does not yet have from the central repo.
+The worker then waits for the raw exposures to arrive at USDF (or confirms they already arrived) and downloads and ingests them.
+When all exposures have arrived, the worker runs the pipeline, updating the APDB and sending alerts as appropriate; if not all exposures arrive, it attempts to process what it has.
+Once pipeline processing is complete, the worker uploads its outputs to the central repository.
+The worker will likely also compute metrics at this stage, but the details (in particular, whether they are computed before or after the central repository sync) have not yet been decided.
+
+For performance reasons, each worker places pipeline outputs from all visits into a small number of Butler runs, which are persisted when each worker uploads to the central repository.
+The run name is a deterministic function of the Science Pipelines configuration, the instrument and choice of pipeline, and the processing date.
+The use of a separate run for each pipeline is required by the Gen 3 provenance system (in particular, the storage of pipeline configuration information and "init-output" datasets), but makes it difficult for the Prompt Processing system to adjust its choice of pipeline in response to missing inputs or to processing failures.
+
+.. _general-priorities:
+
+General Priorities
+==================
+
+Of the outputs produced by the Alert Production pipeline, the most important to have correct and uncorrupted is the APDB, which serves as the source of truth both for later processing and for the externally available Prompt Products database.
+The alert stream is a close second, since it is the primary user interface for transient follow-up and cannot be "fixed" after the fact.
```

Comment:
We can reissue or retract alerts, as described below, but the original erroneous alert could have triggered downstream actions.